google/cloud/bigtable/internal/query_plan.h

```diff
@@ -0,0 +1,88 @@
+
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_INTERNAL_QUERY_PLAN_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_BIGTABLE_INTERNAL_QUERY_PLAN_H
+
+#include "google/cloud/bigtable/version.h"
+#include "google/cloud/completion_queue.h"
+#include <google/bigtable/v2/bigtable.pb.h>
+#include <string>
+#include <utility>
+
+namespace google {
+namespace cloud {
+namespace bigtable_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+
+class QueryPlan : public std::enable_shared_from_this<QueryPlan> {
+ public:
+  // Typically, a lambda capturing the original PrepareQueryRequest and
+  // DataConnection pointer necessary to call the PrepareQuery RPC.
+  using RefreshFn = std::function<google::bigtable::v2::PrepareQueryResponse()>;
+
+  explicit QueryPlan(CompletionQueue const& cq,
```

Comment:
We don't want a public constructor for QueryPlan. We always want to call Create.